Add latex rendering

yanglyu902 opened the pull request:
Math equations (both inline "$...$" and block equations "$$...$$") can now be rendered in assistant responses. Two additional packages: `marked` and `marked-katex-extension` are imported, slightly increasing the browser loading time after `npm run dev`.

Notice that we only want to render latex when displaying the content, otherwise thousands of html math tags generated by Katex will be stored as chat output, which should be avoided.

Maybe it's useful for the user to choose whether to include this feature during compile, but I'm not quite sure how to do that...

Niek commented:
This PR adds external dependencies (`katex.min.css` and `tex-mml-chtml.js`), those should be bundled instead. We can make support for LaTeX toggle-able with a .env flag.

yanglyu902 commented:
@Niek I added `katex.min.css` file into `src` folder and imported in `Messages.svelte`; I also added a tag `VITE_RENDER_LATEX` in `.env` to toggle rendering option. Please let me know if anything needs to be changed, thanks!

felixschwamm commented:
I would really love to have this feature in chatgpt-web. It’s nearly impossible not to get confused with the plain text output when it comes to more complex math expressions so at the moment this is not really suited for mathematical purposes. Katex support would fix this issue.

Niek commented:
See #447
